### DISCUSSION TITLE
network unreliable on some containers

### DISCUSSION THREAD

**Discussion by beertje44** [containers/podman#16876] [Q&A]
I have a wodby/php container running a local checkout of a php app. ATM it has unrealiable networking, sometimes it works and sometimes it doesn't. I use netavark because I have a dual stack ipv4/ipv6 network and understood this stack is needed to support that. At start I had 2 podman networks in this container: 1 for general purpose and 1 for internal hosts only.  After I started to notice these problems I converted to just one network for the containers used internally with the option internal set to false.

Unreliable is mostly defined as internal hosts do'n always resolve. I tried ping'ing the database host and most of the times this gives an are your root error for the right ip address (this is wodby specific, it uses an unprivilliged user internally without sudo rights) but every now and then I get an "bad address" for the database host.

My resolv.conf on the container reads:
search dns.podman <my routers domain suffix>
<ipv4 address of podman gateway>
<ipv6 address of podman gateway>
<my ipv4 local address, pi-hole>
<my internet ipv6 address of the host, pi-hole>

To me it seems sometimes it somehow uses the wrong dns server, but I'm unsure how to fix this.

Also the php app is a usenet indexer and it has had problems connecting to its usenet server (timeout) but again not always.

I'm a pretty novice podman user so I'm sure I'm doing something wrong, I just fail to see what :)

**Comment by Luap99** [comment]
maybe related https://github.com/containers/aardvark-dns/issues/151 https://github.com/containers/aardvark-dns/issues/248

**Comment by beertje44** [comment]
I did some further debugging and installed bind-tools in my alpine based docker container. It seems it is using the wrong dns server every now and then. dig gives the right ip address every time for my postgresql host and uses the internal aardvark dns server, host seems to use my hosts dns server (3rd entry) every time and can't find postgresql.

I'm uncertain if I experience the same issue as mentioned above, there are aardvark journalctl entries but not the same and not @time of occurence of this specific problem.

Tried forcing podman to only use the internal dns servers, but that leaves the container without the possibility to resolve exetrnal hosts which is something I do want.

**Comment by beertje44** [comment]
Still suffering from this, I can work around it by assiging for example mysql a dedicated ip on its internal network, but that seems not the way to go on podman or containers in general.... Would rather use a hostname like you are supposed to do.
